webhooks/clusterclass_test.go

```diff
@@ -869,577 +622,435 @@ func TestClusterClassValidation(t *testing.T) {
-				},
-				Spec: clusterv1.ClusterClassSpec{
-					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
-					ControlPlane: clusterv1.ControlPlaneClass{
-						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: ref},
-					},
-					Workers: clusterv1.WorkersClass{
-						MachineDeployments: []clusterv1.MachineDeploymentClass{
-							{
-								Class: "aa",
-								Template: clusterv1.MachineDeploymentClassTemplate{
-									Metadata:       clusterv1.ObjectMeta{},
-									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: ref},
-									Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
-								},
-							},
-						},
-					},
-				},
-			},
-			in: &clusterv1.ClusterClass{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "default",
-				},
-				Spec: clusterv1.ClusterClassSpec{
-					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
-					ControlPlane: clusterv1.ControlPlaneClass{
-						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: ref},
-					},
-					Workers: clusterv1.WorkersClass{
-						MachineDeployments: []clusterv1.MachineDeploymentClass{
-							{
-								Class: "aa",
-								Template: clusterv1.MachineDeploymentClassTemplate{
-									Metadata: clusterv1.ObjectMeta{
-										Labels:      map[string]string{"foo": "bar"},
-										Annotations: map[string]string{"foo": "bar"},
-									},
-									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: incompatibleRef}, // NOTE: this should be tolerated
-									Infrastructure: clusterv1.LocalObjectTemplate{Ref: compatibleRef},
-								},
-							},
-						},
-					},
-				},
-			},
+			old: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+				WithInfrastructureClusterTemplate(
+					builder.InfrastructureClusterTemplate(metav1.NamespaceDefault, "infra1").Build()).
+				WithControlPlaneTemplate(
+					builder.ControlPlaneTemplate(metav1.NamespaceDefault, "cp1").
+						Build()).
+				WithControlPlaneInfrastructureMachineTemplate(
+					builder.InfrastructureMachineTemplate(metav1.NamespaceDefault, "cpInfra1").
+						Build()).
+				WithWorkerMachineDeploymentClass(
+					builder.MachineDeploymentClass("aa").
+						WithInfrastructureTemplate(
+							refToUnstructured(ref)).
+						WithBootstrapTemplate(
+							builder.BootstrapTemplate(metav1.NamespaceDefault, "bootstrap1").Build()).
+						Build()).
+				Build(),
+			in: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+				WithInfrastructureClusterTemplate(
+					builder.InfrastructureClusterTemplate(metav1.NamespaceDefault, "infra1").Build()).
+				WithControlPlaneTemplate(
+					builder.ControlPlaneTemplate(metav1.NamespaceDefault, "cp1").
+						Build()).
+				WithControlPlaneInfrastructureMachineTemplate(
+					builder.InfrastructureMachineTemplate(metav1.NamespaceDefault, "cpInfra1").
+						Build()).
+				WithWorkerMachineDeploymentClass(
+					builder.MachineDeploymentClass("aa").
+						WithInfrastructureTemplate(
+							refToUnstructured(ref)).
```

Comment:
The old test did change the bootstrap ref to incompatibleRef (which is fine for bootstrap templates) and the infra ref to compatibleRef, currently there are no changes in the new test case if I saw correctly